src/vector/mod.rs

```diff
@@ -127,6 +135,66 @@ pub trait VectorOps {
         derivation_path: DerivationPath,
         metadata: &AlloyMetadata,
     ) -> Result<Vec<u8>, AlloyError>;
+
+    /// Rotates vectors from the in-rotation secret for their secret path to the current secret.
+    /// This can also be used to rotate data from one tenant ID to a new one, which most useful when a tenant is
+    /// internally migrated.
+    ///
+    /// WARNINGS:
+    ///     * this involves decrypting then encrypting vectors. Since the vectors are full of floating point numbers,
+    ///       this process is lossy, which will cause some drift over time. If you need perfectly preserved accuracy
+    ///       store the source vector encrypted with `standard` next to the encrypted vector. `standard` decrypt
+    ///       that, `vector` encrypt it again, and replace the encrypted vector with the result.
+    ///     * only one metadata and new tenant ID argument means each call to this needs to have one tenant's vectors.
+    async fn rotate_vectors(
+        &self,
+        encrypted_vectors: EncryptedVectors,
+        metadata: &AlloyMetadata,
+        new_tenant_id: Option<TenantId>,
+    ) -> RotateResult {
+        let decrypt_attempts: Vec<_> = join_all(encrypted_vectors.into_iter().map(
+            |(vector_id, encrypted_vector)| {
+                self.decrypt(encrypted_vector, metadata)
+                    .map(|r| (vector_id, r))
```

Comment:
From the outside this seems like it'd be a good idea. It changes the performance characteristics a bit because instead of doing all the decrypts and waiting then doing all the encrypts you do each decrypt/encrypt as a unit and `join_all` on those, but that should be more performant anyways.